src/util/tab-events.js

```diff
@@ -1,57 +1,59 @@
 import eventToPromise from './event-to-promise'
 
+function tabChanged(tabId) {
+    return new Promise((resolve, reject) => {
+        browser.webNavigation.onCommitted.addListener((details)=>{
+            if(details.tabId == tabId && details.frameId == 0)
+                reject();
```

Comment:
Instead of rejecting, it would be more logical to resolve here, since the name is tabChanged (I'd call it `whenTabChanged` for consistency, or an even more accurate description). Where it is used code would then switch from `catch` to `then`.